src/SonarScanner.MSBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -39,58 +39,36 @@ public class PreprocessorObjectFactory : IPreprocessorObjectFactory
         public PreprocessorObjectFactory(ILogger logger) =>
             this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
 
-        public async Task<ISonarWebServer> CreateSonarWebServer(ProcessedArgs args, IDownloader downloader = null, IDownloader apiDownloader = null)
+        public async Task<ISonarWebServer> CreateSonarWebServer(ProcessedArgs args, IDownloader webDownloader = null, IDownloader apiDownloader = null)
         {
             _ = args ?? throw new ArgumentNullException(nameof(args));
             var userName = args.GetSetting(SonarProperties.SonarToken, null) ?? args.GetSetting(SonarProperties.SonarUserName, null);
             var password = args.GetSetting(SonarProperties.SonarPassword, null);
             var clientCertPath = args.GetSetting(SonarProperties.ClientCertPath, null);
             var clientCertPassword = args.GetSetting(SonarProperties.ClientCertPassword, null);
 
-            if (!Uri.IsWellFormedUriString(args.ServerInfo.ServerUrl, UriKind.Absolute))
+            if (!ValidateServerUrl(args.ServerInfo.ServerUrl))
```

Comment:
This is the cleanup you asked for in the previous PR.
Let me know if you have any more ideas.